airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -512,6 +513,36 @@ def get_job(self, job_name: str, namespace: str) -> V1Job:
         """
         return self.batch_v1_client.read_namespaced_job(name=job_name, namespace=namespace, pretty=True)
 
+    def get_job_status(self, job_name: str, namespace: str) -> V1Job:
+        """Get job with status of specified name from Google Cloud.
+
+        :param job_name: Name of Job to fetch.
+        :param namespace: Namespace of the Job.
+        :return: Job object
+        """
+        return self.batch_v1_client.read_namespaced_job_status(
+            name=job_name, namespace=namespace, pretty=True
+        )
+
+    def wait_job_until_complete(self, job_name: str, namespace: str, poll_interval: float = 10) -> V1Job:
+        """Block job of specified name from Google Cloud until it is complete.
+
+        :param job_name: Name of Job to fetch.
+        :param namespace: Namespace of the Job.
+        :param poll_interval: Interval in seconds between polling the job status
+        :return: Job object
+        """
+        while True:
+            self.log.info("Requesting status for the job '%s' ", job_name)
+            job: V1Job = self.get_job_status(job_name=job_name, namespace=namespace)
+            if conditions := job.status.conditions:
+                if condition_complete := next((c for c in conditions if c.type == "Complete"), None):
+                    if condition_complete.status:
```

Comment:
Why not simply `if condition_complete` ? Why check the status?

Reply:
Because the `status` field can be `True` (job completed) or `False` (job incomplete).

Reply:
In that case maybe it's better to use `status` instead of `type` on line 539? I just don't get why we need two checks.

Reply:
The `job.status.conditions` is a list of dictionaries with different types. The method tracks on condition with type `Complete` and the status `True`. Only this combination of values ensures that a kubernetes job is completed.
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/job-v1/#JobStatus

Reply:
Ok, so that's some Kubernetes weirdness then. Thanks for the explanation! I have a suggestion about making it a bit more readable, but it is less optimized, since it will have to loop through every item. 

`if any([c for c in conditions if c.type == "Complete" and c.status is True]): `